src/yunohost/tools.py

```diff
@@ -555,3 +556,22 @@ def tools_diagnosis(auth, private=False):
         diagnosis['private']['domains'] = domain_list(auth)['domains']
 
     return diagnosis
+
+
+def tools_portavailable(port):
+    """
+    Check availability of a local port
+
+    Keyword argument:
+        port -- Port to check
+
+    """
+    try:
+        s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        s.settimeout(1)
+        s.connect(("localhost", int(port)))
```

Comment:
About ipv6 : localhost should resolve correctly even if we're in ipv6 (I assume it depends of the content of /etc/hosts, but by default there's the ipv6 equivalent of 127.0.0.1 on most setups I believe...)

And if a service is down : well... then we go back to the point brought my Bram : the clean way would be that each time a program needs a port, it should 'book' it. But that's quite a lot of work and beyond the scope of this PR which was just meant to fix a semantic issue.